pre_commit_hooks/trailing_whitespace_fixer.py

```diff
@@ -78,17 +77,17 @@ def main(argv=None):  # type: (Optional[Sequence[str]]) -> int
 
     # reject probable "eaten" filename as extension: skip leading '.' with [1:]
     for ext in md_exts:
-        if any(c in ext[1:] for c in r'./\:'):
+        if any(c in ext[1:] for c in r'/\:'):
             parser.error(
-                'bad --markdown-linebreak-ext extension {!r} (has . / \\ :)\n'
+                'bad --markdown-linebreak-ext extension {!r} (has / \\ :)\n'
                 "  (probably filename; use '--markdown-linebreak-ext=EXT')"
                 .format(ext),
             )
     chars = None if args.chars is None else args.chars.encode('utf-8')
     return_code = 0
     for filename in args.filenames:
-        _, extension = os.path.splitext(filename.lower())
-        md = all_markdown or extension in md_exts
+        in_md_exts = any([filename for ext in md_exts if filename.endswith(ext)])
```

Comment:
this breaks for the reason I listed here: https://github.com/pre-commit/pre-commit-hooks/issues/445#issuecomment-582520693

Reply:
I don't think so, please add a test to show this behaviour.

Reply:
yes, please do so

Reply:
:+1: